### ISSUE TITLE
`session.s3_is_overloaded` leaks credentials via logging during server timeouts

### ISSUE THREAD

**Issue by HebaruSan** [jjjake/internetarchive#477]
## Problem

Our project has configured `logging.error` to print messages to our public Discord server. We just started getting a lot of these:

![image](https://user-images.githubusercontent.com/1559108/152813382-e74980fd-87d3-4ef9-a90d-cbf1b9ea81c6.png)

Each of those blurs is a copy of our IA access key. The secret key isn't there, so we are not completely compromised, but we'd still prefer to keep both tokens confidential and will have to do some clean-up after this incident.

(Yes, we are reconsidering how our log handler works, but this seemed worth reporting upstream.)

## Cause

It seem that `internarchive.ArchiveSession.send` is printing these here:

https://github.com/jjjake/internetarchive/blob/7330ef77814a57a52a8d0b912238ce8397d38d03/internetarchive/session.py#L549-L561

And the URL containing the access key is constructed here:

https://github.com/jjjake/internetarchive/blob/7330ef77814a57a52a8d0b912238ce8397d38d03/internetarchive/session.py#L338-L345

## Suggestions

- Hide the credentials by using POST instead of GET (I don't know whether archive.org supports this)
- Allow calling code to control whether and how these messages are printed somehow (perhaps exceptions that could be caught, or a configuration parameter somewhere?)
- Maybe `logging.warning` could be used instead (this would only help for log handlers that ignore warnings as ours does)? The `s3_is_overloaded` call doesn't treat this as a serious problem but just discards the exception and returns `True`.

Whatever the means of getting there, the access key shouldn't be printed into logs by default.

FYI to @DasSkelett.

**Comment by JustAnotherArchivist** [comment]
FWIW, the access key isn't considered private anyway. Anyone can see which access key was used to upload a file to IA, for example.

**Comment by HebaruSan** [comment]
> FWIW, the access key isn't considered private anyway. Anyone can see which access key was used to upload a file to IA, for example.

Do you know of any documentation that clarifies that explicitly? I looked for some but didn't find anything.

**Comment by JustAnotherArchivist** [comment]
I don't think so, but then again, many details of IA are undocumented. What I'm referring to simply contains the auth header supplied on the upload. The secret key is explicitly redacted while the access key is left alone. Since that's more complicated to do than just wiping the header altogether, it has to be intentional. But an official statement on that would be better, yeah.

**Comment by HebaruSan** [comment]
> This is intentional. The access key is not considered private, and it is useful for debugging. Thank you for the report, and please let me know if you have any concerns about this.

Thanks! Good to know that we don't need to worry. Same question as before though: Is there some official source that says this? And/or, can you show me how to view access keys for downloads? I tried printing the HTTP response headers for some of the files we've uploaded, and it wasn't there.

**Comment by jjjake** [comment]
I don't believe there is an official source, but you can write info@archive.org if you'd like to confirm.

To view your access key, you can look in your ia config file or visit https://archive.org/account/s3.php while logged in to your account.

**Comment by HebaruSan** [comment]
> To view your access key, you can look in your ia config file or visit https://archive.org/account/s3.php while logged in to your account.

Oh, what I meant was viewing _other users'_ access keys, as @JustAnotherArchivist seemed to suggest (I know where my own keys are already :laughing:):

> What I'm referring to simply contains the auth header supplied on the upload. The secret key is explicitly redacted while the access key is left alone.

I was trying to figure out what that refers to. It kind of sounded like HTTP reponse headers, but access keys do not seem to be included there. If it's possible to view other users' access keys, that would go a long way to confirming that they're not supposed to be private.

**Comment by HebaruSan** [comment]
I'm guessing that whoever monitors that email would not be prepared to answer questions about previous comments in this conversation, but thanks!

**Comment by jjjake** [comment]
@HebaruSan They will be able to help you, or put you in touch with someone who can.
